scalameta/contrib/shared/src/main/scala/scala/meta/contrib/ScaladocToken.scala

```diff
@@ -0,0 +1,197 @@
+package scala.meta.contrib
```

Comment:
a few comments, after reading your review of the other implementation several years ago (and some online, in subsequent code).

while this does look a tree, i didn't understand your suggestion referring to @root and @leaf, so i didn't implement that; i looked at the Adt code but didn't really understand what it does, and have zero experience with macros